tests/php/Unit/Analyzer/TupleSymbol/ForeachSymbolTest.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhelTest\Unit\Analyzer\TupleSymbol;
+
+use Phel\Analyzer;
+use Phel\Analyzer\TupleSymbol\ForeachSymbol;
+use Phel\Ast\ForeachNode;
+use Phel\Exceptions\PhelCodeException;
+use Phel\GlobalEnvironment;
+use Phel\Lang\Symbol;
+use Phel\Lang\Tuple;
+use Phel\NodeEnvironment;
+use PHPUnit\Framework\TestCase;
+
+final class ForeachSymbolTest extends TestCase
+{
+    public function testRequiresAtLeastTwoArg(): void
+    {
+        $this->expectException(PhelCodeException::class);
+        $this->expectExceptionMessage("At least two arguments are required for 'foreach");
+
+        // This is the same as: (foreach)
+        $tuple = Tuple::create(
+            Symbol::create(Symbol::NAME_FOREACH)
+        );
+
+        $this->analyze($tuple);
+    }
+
+    public function testFirstArgMustBeATuple(): void
+    {
+        $this->expectException(PhelCodeException::class);
+        $this->expectExceptionMessage("First argument of 'foreach must be a tuple.");
+
+        // This is the same as: (foreach x)
+        $mainTuple = Tuple::create(
+            Symbol::create(Symbol::NAME_FOREACH),
+            Symbol::create('x')
+        );
+
+        $this->analyze($mainTuple);
+    }
+
+    public function testArgForTupleCanNotBe1(): void
+    {
+        $this->expectException(PhelCodeException::class);
+        $this->expectExceptionMessage("Tuple of 'foreach must have exactly two or three elements.");
+
+        // (foreach [x])
+        $mainTuple = Tuple::create(
+            Symbol::create(Symbol::NAME_FOREACH),
+            Tuple::createBracket(
+                Symbol::create('x')
+            )
+        );
+
+        $this->analyze($mainTuple);
+    }
+
+    public function testValueSymbolFromTupleWith2Args(): void
+    {
+        // (foreach [x []])
+        $mainTuple = Tuple::create(
+            Symbol::create(Symbol::NAME_FOREACH),
+            Tuple::createBracket(
+                Symbol::create('x'),
+                Tuple::createBracket()
+            ),
+            Tuple::create()
+        );
+
+        $node = $this->analyze($mainTuple);
+        self::assertEquals(Symbol::create('x'), $node->getValueSymbol());
+    }
+
+    public function testValueSymbolFromTupleWith3Args(): void
+    {
+        // (foreach [key value {}])
+        $mainTuple = Tuple::create(
+            Symbol::create(Symbol::NAME_FOREACH),
+            Tuple::createBracket(
+                Symbol::create('key'),
+                Symbol::create('value'),
+                Tuple::createBracket()
```

Comment:
It doesn't matter for the test. But to keep it correct, this must be `Table::empty()` and the comment above must be `// (foreach [key value @{}])` (`@` is missing). The same applies to the the next test.

Reply:
Done